core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3506,6 +3510,16 @@ public PhysicalOperation visitExchange(ExchangeNode node, LocalExecutionPlanCont
             return createLocalExchange(node, context);
         }
 
+        private boolean isLocalScaledWriterExchange(PlanNode node)
```

Comment:
use `PlanNodeSearcher` instead (see `io.trino.operator.join.JoinUtils#isBuildSideReplicated`)

Reply:
How about we add scale flag in `TableWriterNode`? That way we don't have to search anything here